sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+    public Mono<DocumentResultCollection<AnalyzeSentimentResult>> analyzeBatchSentiment(
+        List<TextDocumentInput> textInputs) {
+        try {
+            return analyzeBatchSentimentWithResponse(textInputs, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Returns a sentiment prediction, as well as sentiment scores for each sentiment class (Positive, Negative, and
+     * Neutral) for the document and each sentence within it.
+     *
+     * @param textInputs A list of {@link TextDocumentInput inputs/documents}  to be analyzed.
+     * @param options The {@link TextAnalyticsRequestOptions options} to configure the scoring model for documents
+     * and show statistics.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the {@link
+     * DocumentResultCollection batch} of {@link AnalyzeSentimentResult text sentiment}.
+     *
+     * @throws NullPointerException if {@code textInputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<AnalyzeSentimentResult>>> analyzeBatchSentimentWithResponse(
+        List<TextDocumentInput> textInputs, TextAnalyticsRequestOptions options) {
+        try {
+            return withContext(context -> analyzeBatchSentimentWithResponse(textInputs, options, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<AnalyzeSentimentResult>>> analyzeBatchSentimentWithResponse(
+        List<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
+        final MultiLanguageBatchInput batchInput = new MultiLanguageBatchInput()
+            .setDocuments(convertToMultiLanguageInput(documents));
+        return service.sentimentWithRestResponseAsync(
+            batchInput,
+            options == null ? null : options.getModelVersion(),
+            options == null ? null : options.showStatistics(), context)
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of text sentiment input - {}", ignoredValue))
+            .doOnSuccess(response -> logger.info("A batch of text sentiment output - {}", response))
+            .doOnError(error -> logger.warning("Failed to text sentiment - {}", error))
+            .map(response -> new SimpleResponse<>(response, toDocumentResultCollection(response.getValue())));
+    }
+
+    /**
+     * Get default country hint code.
+     *
+     * @return the default country hint code
+     */
+    public String getDefaultCountryHint() {
+        return defaultCountryHint;
+    }
+
+    /**
+     * Get default language when the builder is setup.
+     *
+     * @return the default language
+     */
+    public String getDefaultLanguage() {
+        return defaultLanguage;
+    }
```

Comment:
Could we move these getters up with `getServiceVersion`? It's easier to find getters if they are grouped together.